plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -457,17 +464,42 @@ public static ColumnMapping timestampColumnMapping()
 
     public static ColumnMapping timestampColumnMapping(TimestampType timestampType)
     {
-        checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return ColumnMapping.longMapping(
+        if (timestampType.getPrecision() <= MAX_SHORT_PRECISION) {
+            return ColumnMapping.longMapping(
+                    timestampType,
+                    timestampReadFunction(timestampType),
+                    timestampWriteFunction(timestampType));
+        }
+        checkArgument(timestampType.getPrecision() <= MAX_LOCALDATETIME_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        return ColumnMapping.objectMapping(
                 timestampType,
-                timestampReadFunction(timestampType),
-                timestampWriteFunction(timestampType));
+                longTimestampReadFunction(timestampType),
+                longTimestampWriteFunction(timestampType));
     }
 
     public static LongReadFunction timestampReadFunction(TimestampType timestampType)
     {
         checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
-        return (resultSet, columnIndex) -> toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+        return (resultSet, columnIndex) -> {
+            long epochMicros = toPrestoTimestamp(timestampType, resultSet.getObject(columnIndex, LocalDateTime.class));
+            verify(epochMicros == round(epochMicros, MAX_SHORT_PRECISION - timestampType.getPrecision()),
```

Comment:
Also, why not make the verification part of `toPrestoTimestamp` method? Would avoid duplicating the verify here.

Reply:
Done!